src/coreclr/tools/aot/ILCompiler.Compiler/Compiler/DependencyAnalysis/ObjectWriter.cs

```diff
@@ -1010,6 +1007,12 @@ public static void EmitObject(string objectFilePath, IEnumerable<DependencyNode>
                     objectWriter.SetSection(section);
                     objectWriter.EmitAlignment(nodeContents.Alignment);
 
+                    int neededInterruptionsBytes = nodeContents.Data.Length + 1;
+                    if (objectWriter._byteInterruptionOffsets == null || objectWriter._byteInterruptionOffsets.Length < neededInterruptionsBytes)
+                        objectWriter._byteInterruptionOffsets = new bool[nodeContents.Data.Length + 1];
+                    else
+                        Array.Fill(objectWriter._byteInterruptionOffsets, false);
```

Comment:
This can be pretty wasteful if there is a large node followed by a lot of small nodes.

Reply:
Yeah, I already realized because it got pretty slow. We have a 400 kB node in the metadata blob :)